_posts/2021-01-27-#20833.md

```diff
@@ -0,0 +1,140 @@
+for validity.
+On master, `testmempoolaccept` is limited to one transaction at a time, which means users cannot
+test transactions that depend on outputs that the mempool has not seen yet.
+The obstacle is that our validation code uses a `CoinsViewMemPool`, which tracks the coins
+available in the current chain state and mempool (and not anything else), to validate transaction
+inputs.
+
+- Another reason package mempool accept may be of interest is improving fee bumping.
+Currently, Child Pays for Parent does not work for transactions with a feerate below the mempool
+minimum fee, which can be dangerous for recipients of time-sensitive transactions.
+[PR#20833](https://github.com/bitcoin/bitcoin/pull/20833) is a first step towards package
+validation and hopefully, eventually, Package Relay.
+You can read more about Package Mempool Acceptance
+[here](https://brink.dev/blog/2021/01/21/fellowship-project-package-accept/).
+
+
+## Notes
+
+- [#18033](https://github.com/bitcoin/bitcoin/pull/20833) enables dry runs of package validation.
+It does this by having `CCoinsViewMemPool` keep track of the `Coins` created and spent by
+transactions in the package.
+
+- The [`MemPoolAccept`](https://github.com/bitcoin/bitcoin/blob/7f653c3b22f0a5267822eec017aea6a16752c597/src/validation.cpp#L451)
+ class represents a "validation session" with the current mempool and chain
+state; it holds `cs_main` and `pool.cs` mutexes throughout validation.
+Its job is to figure out whether a transaction (received from peers or clients like RPC) is a valid
+candidate for submission to the mempool.
+To do so, it applies consensus and
+[policy](https://github.com/bitcoin/bitcoin/tree/c831e105/src/policy) rules.
+Importantly, if a transaction is invalid, `MemPoolAccept` is designed such that it fails as quickly
+as possible to mitigate DoS risks.
+
+- Each `MemPoolAccept` instance creates a `CoinsView` from the current chain tip and mempool.
+`CCoinsView` is a class with a hierarchical structure (see diagram below).
+
+```
+             +--------------------+
+             |                    |
+             |     CCoinsView     |
+             |                    |
+             +----------+---------+
+                        ^
+                        |
+             +----------+---------+
+             |                    |
+             |  CCoinsViewBacked  |
+             |                    |
+             +------+------+------+
+                    ^      ^
+           +--------+      +--------+
+           |                        |
++----------+---------+  +-----------+---------+
+|                    |  |                     |
+|  CCoinsViewCache   |  |  CCoinsViewMemPool  |
+|                    |  |                     |
++--------------------+  +---------------------+
+```
+
+- `CoinsViewBacked`s can be "layered" by setting one's `Backend` to another. For example, when a
```

Comment:
maybe
- replace the "s" after `CoinsViewBacked` with a noun for the type of object or data structure
- explain what `Backend` is before using it (and how it can be set to another one, or what that does or what it means)